provider/role_data_test.go

```diff
@@ -0,0 +1,186 @@
+package provider
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/firehydrant/terraform-provider-firehydrant/firehydrant"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func TestAccRoleDataSource_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRoleDataSourceConfig_basic(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Test data source lookup by slug
+					resource.TestCheckResourceAttrSet("data.firehydrant_role.member", "id"),
+					resource.TestCheckResourceAttr("data.firehydrant_role.member", "slug", "member"),
+					resource.TestCheckResourceAttrSet("data.firehydrant_role.member", "name"),
+					resource.TestCheckResourceAttrSet("data.firehydrant_role.member", "description"),
+
+					resource.TestCheckResourceAttrSet("data.firehydrant_role.member", "permissions.#"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccRoleDataSource_builtIn(t *testing.T) {
+	// Test looking up a built-in role that should always exist
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBuiltInRoleDataSourceConfig(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.firehydrant_role.member", "id"),
+					resource.TestCheckResourceAttr("data.firehydrant_role.member", "slug", "member"),
+					resource.TestCheckResourceAttrSet("data.firehydrant_role.member", "permissions.#"),
+				),
+			},
+		},
+	})
+}
+
+func testAccRoleDataSourceConfig_basic() string {
+	return `
```

Comment:
I like the idea here, but we're literally testing the same thing twice, so the extra test doesn't really add value.  If we want to get some value here (and that might be nice for a custom role), we can add one in the terraform CI org (ops+terraform-ci@firehydrant.io in the Engineering Secrets vault in 1 pass) to make sure this is fetchable for the live test.